src/app/admin/product-form/product-form.component.html

```diff
@@ -0,0 +1,81 @@
+<form #f="ngForm" (ngSubmit)="save(f.value)">
+  <div class="form-group">
+    <label for="title">Title</label>
+    <input
+      #title="ngModel"
+      ngModel
+      name="title"
+      id="title"
+      type="text"
+      class="form-control"
+      required
+    />
+    <div class="alert alert-danger" *ngIf="title.touched && title.invalid">
+      Title is required.
+    </div>
+  </div>
+  <div class="form-group">
+    <label for="price">Price</label>
+    <div class="input-group mb-3">
+      <div class="input-group-prepend">
+        <span class="input-group-text">$</span>
+        <input
+          #price="ngModel"
+          ngModel
+          name="price"
+          id="price"
+          type="number"
+          class="form-control"
+          required
+        />
+      </div>
+      <div class="alert alert-danger" *ngIf="price.touched && price.invalid">
+        Price is required.
+      </div>
+    </div>
+  </div>
+  <div class="form-group">
+    <label for="category">Category</label>
+    <select
+      #category="ngModel"
+      ngModel
+      name="category"
+      id="category"
+      class="form-control"
+      required
+    >
+      <option value=""></option>
+      <option
+        *ngFor="let category of categoriesObservable | async"
+        [value]="category.key"
```

Comment:
My code has category.$key. But if it works without the $ it should be fine.

Reply:
yea does this work without the $

Reply:
Everything seems to function properly with it so far.